File: 3th_Course/Q2/Data_Procesing/Unit_1/JavaVSscala/ex06.scala
class Rectangle(val width: Double, val height: Double) {
  // Método para calcular el área
  def area(): Double = {
    return width * height
  }
}

// Llamada en Scala (versión traducida)
val rect = new Rectangle(5.0, 10.0)
println(s"Area: ${rect.area()}")


// en scala puro

class Rectangle(val width: Double, val height: Double) {
  // Método para calcular el área (sin return)
  def area: Double = width * height
}

// Llamada en Scala (versión idiomática)
val rect = new Rectangle(5.0, 10.0)
println(s"Area: ${rect.area}")

File: 3th_Course/Q2/Data_Procesing/Unit_1/JavaVSscala/ex06.java
public class Rectangle {
    private final double width;
    private final double height;

    // Constructor
    public Rectangle(double width, double height) {
        this.width = width;
        this.height = height;
    }

// Método para calcular el área
    public double area() {
        return width * height;
    }

    // Getters
    public double getWidth() {
        return width;
    }

    public double getHeight() {
        return height;
    }
}

// Llamada en Java
Rectangle rect = new Rectangle(5.0, 10.0);
System.out.println("Area: " + rect.area());